resource/compute_dynamic_params.py

```diff
@@ -0,0 +1,182 @@
+import logging
+from dataiku import SQLExecutor2, api_client, default_project_key
+
+
+def do(payload, config, plugin_config, inputs):
+    """
+    Currently we can't dynamically recalculate the choices of a param B depending on the value of a param A (see https://app.clubhouse.io/dataiku/story/53713)
+    This is why we have this tradeoff for macros created from a scenario: list all the Snowflake objects, grouped by connection,
+    rather than only the ones from the connection of the selected dataset.
+    Macros directly created from a dataset currently don't let the user change the input dataset, and list the Snowflake objects from the connection of this
+    dataset.
+    """
+
+    # if the macro is run from the flow, the input_dataset param is set...
+    dataset_name = config.get('input_dataset')
+
+    # ...thus we know whether we should display the macro in "flow" or "scenario" style
+    if dataset_name:
+        return macro_from_dataset_params(payload['parameterName'], dataset_name)
+    else:
+        return macro_from_scenario_params(payload['parameterName'])
+
+
+def macro_from_scenario_params(parameter_name):
+    """
+    Params of type `SELECT` don't let us group the choices by connection. This is why we use this hack of displaying the connections as selectable
+    choices and indenting to the right the valid choices. If the user was to select a connection rather than a valid choice, he would then get an error.
+    """
+
+    # We start by fetching all snowflake datasets in the current project and group them per connection
+    datasets_per_connection = dict()
+    for dataset in get_snowflake_datasets():
+        connection = dataset['params']['connection']
+        if connection not in datasets_per_connection:
+            datasets_per_connection[connection] = []
+        datasets_per_connection[connection] += [dataset.name]
+
+    # Only if there are multiple Snowflake connections, we need to display the different inputs grouped by connections
+    multiple_connections = len(datasets_per_connection) > 1
+
+    # indent only if we have multiple Snowflake connections
+    indent = (lambda choice: {'value': choice['value'], 'label': f"  {choice['label']}"}) if multiple_connections else (lambda choice: choice)
```

Comment:
Not a big fan of this kind of structure hard to read. I would rather right that:
```suggestion
    def do_indentation(choice):
        return {'value': choice['value'], 'label': f"  {choice['label']}"}
    indent = do_indentation if multiple_connections else (lambda choice: choice)
```